.github/workflows/image.yaml

```diff
@@ -40,3 +40,51 @@ jobs:
     secrets:
       registry_username: ${{ github.actor }}
       registry_password: ${{ secrets.CR_PAT }}
+
+  generate-init-kyverno-provenance:
+    needs: push-init-kyverno
+    permissions:
+      id-token: write # To sign the provenance.
+      packages: write # To upload assets to release.
+      actions: read #To read the workflow path.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@main
```

Comment:
In these statements we should pin to a specific version by referencing the cryptographic digest and not point to just the `main` branch, which can change at any moment.

Reply:
The [slsa-github-generator](https://github.com/slsa-framework/slsa-github-generator) team doesn't have a [tagged release](https://github.com/slsa-framework/slsa-github-generator/releases) yet for the container generator.

Reply:
or maybe I can reference the latest commit in the main?

Reply:
See if it'll pass if you do that.

Reply:
I have tested and it's working.
Test Workflow Link: https://github.com/zurrehma/kyverno/actions/runs/3199370933

Reply:
updated the code with the latest commit from main branch